lib/controllers/orders.js

```diff
@@ -27,9 +27,32 @@ module.exports = Router()
       const order = await Order.getById(id);
 
       res.send(order);
-    } catch (err) {
+    } catch(err) {
       next(err);
     }
   })
-  .put('/:id', async (req, res, next) => {})
-  .delete('/:id', async (req, res, next) => {});
+  .put('/:id', async (req, res, next) => {
+    try {
+      const id = req.params.id;
+      const quantity = req.params.quantity;
```

Comment:
This line is the main one that's giving you problems - the `quantity` doesn't live on `req.params`; it lives on `req.body`. This line, then, is always going to set `quantity` to `undefined`. When it hits your `updateById` method, there's that check to see if a quantity was passed through. This will fail that check, and the order's original quantity will be passed along to the sql query instead.